main.go

```diff
@@ -185,3 +223,85 @@ func setupWebhooks(mgr ctrl.Manager) {
 func concurrency(c int) controller.Options {
 	return controller.Options{MaxConcurrentReconciles: c}
 }
+
+// Check if an array contains a value.
+func contains(s []string, str string) (int, bool) {
+	for i, v := range s {
+		if v == str {
+			return i, true
+		}
+	}
+
+	return -1, false
+}
+
+// GetTLSOptionOverrideFuncs returns a list of TLS configuration overrides to be used
+// by the webhook server.
+func GetTLSOptionOverrideFuncs(options TLSOptions) ([]func(*tls.Config), error) {
+	var tlsOptions []func(config *tls.Config)
+	TLSMinVersionIndex, _ := contains(cliflag.TLSPossibleVersions(), options.TLSMinVersion)
+	TLSMaxVersionIndex, _ := contains(cliflag.TLSPossibleVersions(), options.TLSMaxVersion)
+	// Check if tls min version flag value is greater than max version
+	if TLSMinVersionIndex > TLSMaxVersionIndex &&
+		TLSMaxVersionIndex != -1 &&
+		TLSMinVersionIndex != -1 {
+		return nil, fmt.Errorf("cipher suite flag min version is greater than max version")
+	}
+
+	tlsMinVersion, err := cliflag.TLSVersion(options.TLSMinVersion)
+	if err != nil {
+		return nil, err
+	}
+	tlsOptions = append(tlsOptions, func(cfg *tls.Config) {
+		cfg.MinVersion = tlsMinVersion
+	})
+
+	tlsMaxVersion, err := cliflag.TLSVersion(options.TLSMaxVersion)
+	if err != nil {
+		return nil, err
+	}
+	tlsOptions = append(tlsOptions, func(cfg *tls.Config) {
+		cfg.MaxVersion = tlsMaxVersion
+	})
+
+	if options.TLSCipherSuites != "" {
+		tlsCipherSuites := strings.Split(options.TLSCipherSuites, ",")
+		suites, err := cliflag.TLSCipherSuites(tlsCipherSuites)
+		if err != nil {
+			return nil, err
+		}
+
+		if options.TLSMinVersion == "VersionTLS13" {
+			tls13Suites := []string{"TLS_AES_128_GCM_SHA256", "TLS_AES_256_GCM_SHA384", "TLS_CHACHA20_POLY1305_SHA256"}
+
+			for _, cipherSuite := range tlsCipherSuites {
+				_, containstls13Suite := contains(tls13Suites, cipherSuite)
+				if !containstls13Suite {
+					return nil, fmt.Errorf("invalid cipher suite provided for tls version 13")
+				}
+			}
+		}
+		insecureCipherValues := cliflag.InsecureTLSCipherNames()
+		for _, cipher := range tlsCipherSuites {
+			_, containsInsecureCipher := contains(insecureCipherValues, cipher)
+			if containsInsecureCipher {
+				return nil, fmt.Errorf("cipher suite %s is insecure please update", cipher)
+			}
+		}
```

Comment:
I think it would be nicer to get rid of the custom `contains` function and do it like here
https://github.com/kubernetes/kubernetes/blob/93ec57734fbb97b33d6060af834a1dfe045fd914/staging/src/k8s.io/apiserver/pkg/server/secure_serving.go#L63-L73
I would also say that we should not error out because the cipher suite is insecure, we should just log a warning (as in the link).